packages/astro-sst/src/adapter.ts

```diff
@@ -88,6 +87,7 @@ export default function createIntegration(
             updateConfig({
               output: "server",
             });
+            config.output = "server";
```

Comment:
FWIW, this is ugly and I have a PR with Astro open to make it so the config can be returned after an update to the config: https://github.com/withastro/astro/pull/9013